ui/src/components/NodePagePodsTab.spec.tsx

```diff
@@ -0,0 +1,124 @@
+import { brand } from '@scality/core-ui/dist/style/theme';
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import { IntlProvider } from 'react-intl';
+import { QueryClient, QueryClientProvider } from 'react-query';
+import { useSelector } from 'react-redux';
+import { ThemeProvider } from 'styled-components';
+import { mockOffsetSize } from '../tests/mocks/util';
+import translations_en from '../translations/en.json';
+import NodePagePodsTab from './NodePagePodsTab';
+
+const mockUseSelector = useSelector as jest.Mock;
+
+const mockResult = (numContainerRunning: number, status = 'Running') => ({
+  name: 'test-pod',
+  age: '4d2h',
+  namespace: 'xcore',
+  status: {
+    status,
+    numContainer: 3,
+    numContainerRunning,
+  },
+  log: 'test',
+});
+
+const statusHealthyRGB = 'rgb(10, 173, 166)';
+const statusWarningRGB = 'rgb(248, 243, 43)';
+const statusCriticalRGB = 'rgb(232, 72, 85)';
+
+const selectors = {
+  podName: () => screen.getByText('test-pod'),
+  podAge: () => screen.getByText('4d2h'),
+  podNamespace: () => screen.getByText('xcore'),
+  podStatus: (status: string) => screen.getByText(status),
+};
+
+const CustomWrapper = ({ children }: { children?: React.ReactNode }) => {
+  return (
+    <QueryClientProvider
+      client={
+        new QueryClient({
+          defaultOptions: {
+            queries: {
+              retry: false,
+            },
+          },
+        })
+      }
+    >
+      <IntlProvider locale="en" messages={translations_en}>
+        <ThemeProvider theme={brand}>{children}</ThemeProvider>
+      </IntlProvider>
+    </QueryClientProvider>
+  );
+};
+jest.mock('react-redux', () => ({
+  useSelector: jest.fn(),
+}));
+
+describe('NodePagePodsTab', () => {
+  beforeAll(() => {
+    mockOffsetSize(400, 200);
+  });
+  beforeEach(() => {
+    mockUseSelector.mockImplementation((selector) =>
+      selector({
+        config: {
+          api: {
+            url_grafana: 'http://mock.grafana.url',
+          },
+        },
+      }),
+    );
```

Comment:
I think that we can mocking through msw. This way, if we ever transition from Redux to React-Query, we won't have to rewrite this test.